crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -362,15 +362,55 @@ def nodes
 
       protected
 
+      # If there's separate network cluster, we have touch it before we start upgrade of other
+      # nodes, specificaly we need to evacuate the network routers from the first network node.
+      def prepare_network_node(network_node)
+        return if network_node.upgraded?
+
+        evacuate_network_node(network_node, network_node)
```

Comment:
@rhafer is it important which node is the *controller node* in this call? 
If it were not `network_node`, I'd need to make sure `neutron-ha-tool` is installed. But why would it be important? At least in this first invocation of `prepare_network_node`, all nodes are still running Cloud6